### PR TITLE
Modify rule S6602: Update benchmark

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**
Fixes [NET-568]
https://sonarsource.atlassian.net/browse/NET-568

Related benchmark update:
https://github.com/SonarSource/languages-experimental-tooling/pull/272

## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

[NET-568]: https://sonarsource.atlassian.net/browse/NET-568?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: rules/S6602/metadata.json [hunk 1]
```diff
@@ -21,6 +21,6 @@
   "ruleSpecification": "RSPEC-6602",
   "sqKey": "S6602",
   "scope": "All",
-  "defaultQualityProfiles": ["Sonar way"],
+  "defaultQualityProfiles": [],
```
Comment (CristianAmbrosini) [inline rules/S6602/metadata.json:24]: Should we downgrade it to non-SonarWay also for VB.NET?

File: rules/S6602/resources-dotnet.adoc [hunk 2]
```diff
@@ -11,11 +11,33 @@
 
 [options="header"]
 |===
-| Method | Runtime | Mean | Standard Deviation | Allocated
-| FirstOrDefault | .NET 7.0 | 5.373 ms | 0.1049 ms | 125 KB
-| Find | .NET 7.0 | 1.691 ms | 0.0334 ms | 85.94 KB
-| FirstOrDefault | .NET Framework 4.6.2 | 5.035 ms | 0.0421 ms | 125.38 KB
-| Find | .NET Framework 4.6.2 | 1.779 ms | 0.0107 ms | 86.2 KB
+| Method                      | Runtime              | Categories       | Mean       | StdDev    | Allocated
```
Comment (CristianAmbrosini) [inline rules/S6602/resources-dotnet.adoc:14]:
```suggestion
| Method                      | Runtime              | Categories       | Mean       | Standard Deviation    | Allocated
```

**Review by CristianAmbrosini — CHANGES_REQUESTED** [review]

File: rules/S6602/metadata.json (hunk 1 above)
Comment (martin-strecker-sonarsource) [reply rules/S6602/metadata.json:24]: Yes. I think so. This is the top-level file, and the language-specific `metadata.json` files are empty. So there is nothing to do here, right?

Comment (CristianAmbrosini) [reply rules/S6602/metadata.json:24]: Yeah that's why I asked, this way we are downgrading for both languages and I wanted to be sure it was intentional

**Review by CristianAmbrosini — APPROVED** [review]
LGTM

File: rules/S6602/resources-dotnet.adoc [hunk 3]
```diff
@@ -11,11 +11,33 @@
 
 [options="header"]
 |===
-| Method | Runtime | Mean | Standard Deviation | Allocated
-| FirstOrDefault | .NET 7.0 | 5.373 ms | 0.1049 ms | 125 KB
-| Find | .NET 7.0 | 1.691 ms | 0.0334 ms | 85.94 KB
-| FirstOrDefault | .NET Framework 4.6.2 | 5.035 ms | 0.0421 ms | 125.38 KB
-| Find | .NET Framework 4.6.2 | 1.779 ms | 0.0107 ms | 86.2 KB
+| Method                      | Runtime              | Categories       | Mean       | Standard Deviation    | Allocated
```
[the remark this answers is not in the document]
Comment (martin-strecker-sonarsource) [reply rules/S6602/resources-dotnet.adoc:14]: Yes. This is all a bit unfortunate:
* You can only have multiple baselines if you also use categories
* The different baselines give you "Ratio" and "Memory Ratio" outputs, but we hide these columns
* The category columns can not be hidden by the BenchmarkDotNet configuration

So we end up with the worst of all worlds and the category isn't bringing much value. Let's keep it this way. It is closest to the benchmarkdotnet output and copy/paste friendly. It is also better suited for future changes to the benchmark (like bringing the Ratio columns back).

**Pull request merged**